pkg/sensors/handler.go

```diff
@@ -6,33 +6,70 @@ package sensors
 import (
 	"fmt"
 
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/pkg/policyfilter"
 	"github.com/cilium/tetragon/pkg/tracingpolicy"
 )
 
 type handler struct {
 	// map of sensor collections: name -> collection
 	collections               map[string]collection
 	bpfDir, mapDir, ciliumDir string
+
+	nextPolicyID uint64
+	pfState      *policyfilter.State
 }
 
-func newHandler(bpfDir, mapDir, ciliumDir string) *handler {
+func newHandler(bpfDir, mapDir, ciliumDir string) (*handler, error) {
+	pfState, err := policyfilter.GetState()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize policy filter state: %w", err)
+	}
+
 	return &handler{
 		collections: map[string]collection{},
 		bpfDir:      bpfDir,
 		mapDir:      mapDir,
 		ciliumDir:   ciliumDir,
-	}
+		pfState:     pfState,
+		// NB: policy ids start with 1, so that they can be used for filtering. In policy
+		// filtering code, a policy id of 0 means no filtering.
+		nextPolicyID: 1,
+	}, nil
+}
+
+func (h *handler) allocPolicyID() uint64 {
+	ret := h.nextPolicyID
+	h.nextPolicyID++
+	return ret
 }
 
 func (h *handler) addTracingPolicy(op *tracingPolicyAdd) error {
 	if _, exists := h.collections[op.name]; exists {
 		return fmt.Errorf("failed to add tracing policy %s, a sensor collection with the name already exists", op.name)
 	}
+	tpID := h.allocPolicyID()
+	filterID := policyfilter.PolicyID(0)
+
+	// This is a namespaced policy, so update policy filter state before loading the sensors
+	// NB: the filterID is set to a non-zero value only if we have a namespaced policy and need
+	// to apply filtering.
+	if tpNs, ok := op.tp.(tracingpolicy.TracingPolicyNamespaced); ok {
+		filterID = policyfilter.PolicyID(tpID)
+		if err := h.pfState.AddPolicy(filterID, tpNs.TpNamespace()); err != nil {
+			return err
+		}
+	}
```

Comment:
nit: what if instead of presetting `filterID := policyfilter.PolicyID(0)` , you do usual if else path and set the `filterID := policyfilter.PolicyID(0)` only on the else?

And replace that 0 with a constant something like TracingPolicyNotNamespaced so easy to read and we are not lost with the special zeros?

Reply:
> nit: what if instead of presetting filterID := policyfilter.PolicyID(0) , you do usual if else path and set the filterID := policyfilter.PolicyID(0) only on the else?

I don't think this makes a difference. There is no uninitiated memory in go.
If I write:
```
var filterID policyfilter.PolicyID
```

It will initialize this to the default value which is zero.